sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -227,7 +227,10 @@ def GetHeaders(  # pylint: disable=too-many-statements,too-many-branches
         # else serialize using json dumps method which apart from regular values will serialize None into null
         else:
             # single partitioning uses a string and needs to be turned into a list
-            if isinstance(options["partitionKey"], list) and options["partitionKey"]:
+            is_tuple_or_list = True in (isinstance(options["partitionKey"], list),
```

Comment:
Is there any other way we can think of where customers will pass in the HPK differently?
What if the customer just passes one partition key value out of the different HPK values for a container which is HPK based?

Reply:
They can do so if its encased in a list or tuple. If one were to currently pass a single value object like a number or string, it would end up erroring out when we calculate the EPK value. If we wanted to accept such values some rework would be needed for getting the epk ranges for it, but it makes more sense to simply add that single value to a list/tuple. This check makes it so we don't attempt to get an epk range from a non list/tuple pk value and instead allow the backend to return a bad request error about the pk value. The question would be, do we want/need to have that use case?

Reply:
Understood, makes sense. Guidance is good enough I think.